zipkin-server/src/main/java/zipkin2/server/internal/elasticsearch/ZipkinElasticsearchStorageConfiguration.java

```diff
@@ -137,6 +145,30 @@ SessionProtocol esSessionProtocol(ZipkinElasticsearchStorageProperties es) {
     };
   }
 
+  @Bean @Qualifier(QUALIFIER) @Conditional(BasicAuthRequired.class)
+  BasicCredentials dynamicElasticsearchBasicCredentials(ZipkinElasticsearchStorageProperties es) {
+    if (isEmpty(es.getUsername()) || isEmpty(es.getPassword())) {
+      return new BasicCredentials();
+    }
+    return new BasicCredentials(es.getUsername(), es.getPassword());
+  }
+
+  @Bean @Qualifier(QUALIFIER) @Conditional(DynamicRefreshRequired.class)
+  Consumer<ClientOptionsBuilder> dynamicElasticsearchAuth(ZipkinElasticsearchStorageProperties es,
+    @Qualifier(QUALIFIER) BasicCredentials basicCredentials) {
+    return new Consumer<ClientOptionsBuilder>() {
+      @Override
+      public void accept(final ClientOptionsBuilder client) {
+        ScheduledExecutorService ses = Executors.newSingleThreadScheduledExecutor(
```

Comment:
I think maybe spring already has a scheduler bean? if not, define this as a bean so it can shutdown more coherently.

Reply:
I picked up a scheduler bean.